### DISCUSSION TITLE
default  settings

### DISCUSSION THREAD

**Discussion by gadget78** [ZoeyVid/NPMplus#1310] [Q&A] [answered]
im not sure of how i can alter the default setting of NPM when installing, 
as i want the default to be "drop connection"
for the setting of "What to show when nginx is hit from a unknown host"

**Comment by Zoey2936** [comment]
I think you mean when deploying NPMplus automatically, right? Because otherwise you can do it in the UI.

**Reply by Zoey2936** [reply]
If yes, how does your deployment work?

**Reply by gadget78** [reply]
yes sorry,  i have made a "promox installer"
and i want to control how the default behaviour of said setting is.. 

i do have a "runs once" script in the "prerun" folder, that runs to set some other things up
(like i change the streams template file, so port ranges are supported here by default)
so would be great to know where this config is saved/setup, so i can adjust it within that same script maybe ? .. 


(i need to push this proxmox installer publicly here really
it is public now, but is apart of another installer, which wouldn't appeal to most! )

**Reply by Zoey2936** [reply]
you can make an api request using curl which changes the default value on start, like done in this script which creates a host:
https://github.com/ZoeyVid/NPMplus/blob/develop/rootfs/usr/local/bin/aio.sh

**Reply by Zoey2936** [reply]
> like i change the streams template file, so port ranges are supported here by default

what do you mean exactly? prot ranges should be supported an work with npmplus for streams

**Reply by gadget78** [reply]
> > like i change the streams template file, so port ranges are supported here by default
> 
> what do you mean exactly? prot ranges should be supported an work with npmplus for streams

i have this added to my prerun script,
`sed -i "s/{{ forwarding_port }}/\\\$server_port/g" /app/templates/stream.conf && echo " - stream port template updated to support ranges" || { echo " ** failed to update stream range fix"; true; }`
so that the specific incoming "server_port" is used, and not the static "forwarding_port" .. 


if i update file 
`backend/schema/paths/settings/get.json`
within the same prerun script, so that the `"value": "congratulations",` is `"value": "444",`
would that change the initial default to 444, and not how its starts with `congratulations` setting ? ..

**Reply by Zoey2936** [reply]
No it would not, cirl is the only way to update the database and the files on disc

**Reply by Zoey2936** [reply, marked as answer]
An env will be added with the next release to change the initial value

**Reply by Zoey2936** [reply]
if you want, you can test this alpha/prerelease: https://github.com/ZoeyVid/NPMplus/releases/tag/2025-01-03-alpha (but make a backup first)

**Comment by Zoey2936** [comment]
> i have this added to my prerun script,
`sed -i "s/{{ forwarding_port }}/\\\$server_port/g" /app/templates/stream.conf && echo " - stream port template updated to support ranges" || { echo " ** failed to update stream range fix"; true; }`
so that the specific incoming "server_port" is used, and not the static "forwarding_port" .. 

So you add a port range as incomming port and the forwarding port should be the same as the client connects to? Adding `$server_port` as forwarding port within the ui is regex blocked right? Since with your change forwarding port will always be the same as incomming port, right? Should I maybe change the regex instead so you can directly put in `$server_port` as forwarding port in the ui?

**Reply by Zoey2936** [reply]
I will allow $server_port as a valid input as forwarding port in the ui

**Reply by Zoey2936** [reply]
if you want, you can test this alpha/prerelease: https://github.com/ZoeyVid/NPMplus/releases/tag/2025-01-03-alpha (but make a backup first)
